vm/systemSmartContracts/auction.go

```diff
@@ -139,8 +142,21 @@ func (s *stakingAuctionSC) unJail(args *vmcommon.ContractCallInput) vmcommon.Ret
 		return vmcommon.UserError
 	}
 
+	err := s.eei.UseGas(s.gasCost.MetaChainSystemSCsCost.UnJail)
+	if err != nil {
+		return vmcommon.OutOfGas
+	}
+
 	for _, argument := range args.Arguments {
-		_, _ = s.executeOnStakingSC([]byte("unJail@" + hex.EncodeToString(argument)))
+		err = s.eei.UseGas(s.gasCost.BaseOperationCost.DataCopyPerByte * uint64(len(argument)))
```

Comment:
Maybe add some TODOs to evaluate how large will be the final cost of this function. Valid for all operations.

Reply:
that's why we have a config :D